File: core/src/org/rexellentgames/dungeon/entity/item/accessory/equipable/BloodRing.java
package org.rexellentgames.dungeon.entity.item.accessory.equipable;

import org.rexellentgames.dungeon.assets.Locale;
import org.rexellentgames.dungeon.entity.creature.player.Player;

public class BloodRing extends Equipable {
	{
		description = Locale.get("blood_ring_desc");
		name = Locale.get("blood_ring");
		sprite = "item (ring C)";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).regen += 1f;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).regen -= 1f;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/BloodRing.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class BloodRing : Equippable() {
	init {
		description = Locale.get("blood_ring_desc")
		name = Locale.get("blood_ring")
		sprite = "item-ring_c"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.regen += getChance()
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.regen -= getChance()
	}

	private fun getChance(): Float {
		return this.level * 2f - 1
	}
}